### PR TITLE
[helm/charts] [stakater/reloader] Add reloader to helm public charts repo

### PR DESCRIPTION
**Pull request by faizanahmad055**
Signed-off-by: Faizan Ahmad
username: faizanahmad055
email: faizan.ahmad55@outlook.com

#### What this PR does / why we need it:
- [Reloader](https://github.com/stakater/reloader) an opensource tool maintained by [Stakater](https://github.com/stakater)
- Reloader is a Kubernetes controller to watch changes in ConfigMap and Secrets and then restart pods for Deployment, StatefulSet, and DaemonSet

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/reloader/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+apiVersion: extensions/v1beta1
```
Comment (cpanato) [inline stable/reloader/templates/deployment.yaml]: Please use `apps/v1`

Comment (faizanahmad055) [reply stable/reloader/templates/deployment.yaml]: Updated

File: stable/reloader/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "reloader-fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "reloader-labels.chart" -}}
+app: {{ template "reloader-name" . }}
+chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (cpanato) [inline stable/reloader/templates/_helpers.tpl]: please remove the chart and heritage, see https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#fixing-selectors

File: stable/reloader/templates/deployment.yaml (hunk 1 above)
Comment (cpanato) [reply stable/reloader/templates/deployment.yaml]: thanks! :)

File: stable/reloader/templates/_helpers.tpl (hunk 2 above)
Comment (faizanahmad055) [reply stable/reloader/templates/_helpers.tpl]: As mentioned [here](https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#metadata), we also need chart and heritage in main metadata labels. So I've moved `app` and `release` labels to `reloader-labels.selector` function and `chart` and `heritage` labels to `reloader-labels.chart` function. Please take a look.

File: stable/reloader/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
+name: reloader
+description: Reloader chart that runs on kubernetes
+version: 0.0.18
```
Comment (cpanato) [inline stable/reloader/Chart.yaml]: would use `1.0.0` since this chart is in stable and is the first release

**Comment by cpanato** [comment]
just a small, otherwise lgtm

**Comment by cpanato** [comment]
can I have your eyes here as well @davidkarlsen ?

File: stable/reloader/Chart.yaml (hunk 3 above)
Comment (faizanahmad055) [reply stable/reloader/Chart.yaml]: Updated

**Review by cpanato — APPROVED** [review]
lgtm! thanks!

**Comment by faizanahmad055** [comment]
Hi @davidkarlsen, @foxish, @seanknox
Ping

**Comment by faizanahmad055** [comment]
Hi @davidkarlsen, @foxish, @seanknox
Ping

**Comment by faizanahmad055** [comment]
Hi @davidkarlsen, @foxish, @seanknox
Can you guys take a look? All requested changes have been implemented.

**Comment by faizanahmad055** [comment]
Hi @davidkarlsen, @foxish, @seanknox, @cpanato
Can you guys take a look? This is a very useful tool. All change requests have been implemented. I think you just need to merge it.

**Comment by faizanahmad055** [comment]
Hi @davidkarlsen, @foxish, @seanknox, @cpanato
Any update on this, please?

**Review by cpanato — APPROVED** [review]

**Pull request merged**